### PR TITLE
Add getDiscordData() in createDataset.py

### PR DESCRIPTION
**Pull request by TotallyNotChase**
# Changelog
* Added `getDiscordData()` in `createDataset.py` - This maps a dict with `other person's response - > my response` much like the other methods. This simply adds support with **Discord**. Make sure you read the [README](https://github.com/adeshpande3/Facebook-Messenger-Bot/blob/master/README.md) before you use it! You'll need this [tool](https://github.com/Tyrrrz/DiscordChatExporter) to get your discord data 
* Updated README with `getDiscordData()` docs
* Added `.gitignore` - removes `.txt` and `.npy` files
* Changed `personName` to local scope instead of global scope. This allows users to still use this tool if they have different names across different platforms
* Changed all `open(...)` commands in `createDataset.py` to `with open(....)`
* Fixed **formatting**, there were some of *mixed tabs and spaces* in `createDataset.py` - changed to only `space` characters for consistency
* Fixed comment spacing for the comment explaining `responseDict`. Changed to documentation level comment.
* Added a missing `str` cast in `getLinkedData`.

### CODE REVIEW

**Comment by TotallyNotChase** [comment]
closes #50

File: createDataset.py [hunk 1]
```diff
@@ -4,18 +4,144 @@
 import re
 from datetime import datetime
 
-personName = raw_input('Enter your full name: ')
```
Comment (adeshpande3) [inline createDataset.py]: The reason is that we're asking for the name individually in each of the functions, correct?

File: createDataset.py [hunk 2]
```diff
@@ -4,18 +4,144 @@
...
+def getGoogleHangoutsData():
+    personName = raw_input('Enter your full Hangouts name: ')
+    # Putting all the file names in a list
+    allFiles = []
+    # Edit these file and directory names if you have them saved somewhere else
+    for filename in os.listdir('GoogleTextForm'):
+        if filename.endswith(".txt"): 
+            allFiles.append('GoogleTextForm/' + filename)
+
+    responseDictionary = dict() # The key is the other person's message, and the value is my response
```
Comment (adeshpande3) [inline createDataset.py]: Nit: Comment spacing looks off here. Might've been there before, but I think cleaning it up would be good.

File: createDataset.py [hunk 3]
```diff
@@ -25,162 +151,74 @@ def getWhatsAppData():
                     myMessage = cleanMessage(myMessage)
                     responseDictionary[otherPersonsMessage.rstrip()] = myMessage.rstrip()
                     otherPersonsMessage, myMessage = "",""
-                otherPersonsMessage = otherPersonsMessage + str(row['Content']) + " "
```
Comment (adeshpande3) [inline createDataset.py]: What's the reason for this again? Is it a python 2 -> 3 thing?

File: createDataset.py [hunk 4]
```diff
@@ -25,162 +151,74 @@ def getWhatsAppData():
...
-	return responseDictionary
+def getDiscordData():
+    personName = raw_input('Enter your full Discord name: ')
+    # Putting all the file names in a list
+    allFiles = []
+    # Edit these file and directory names if you have them saved somewhere else
+    for filename in os.listdir('DiscordChatLogs'):
+        if filename.endswith(".txt"): 
+            allFiles.append('DiscordChatLogs/' + filename)
+    responseDictionary = dict() # The key is the other person's message, and the value is my response
```
Comment (adeshpande3) [inline createDataset.py]: Nit: Same nit with the comment spacing. If it's just rendering weird on Github then its fine.

File: createDataset.py [hunk 5]
```diff
@@ -25,162 +151,74 @@ def getWhatsAppData():
...
-	# Remove new lines within message
-	cleanedMessage = message.replace('\n',' ').lower()
-	# Deal with some weird tokens
-	cleanedMessage = cleanedMessage.replace("\xc2\xa0", "")
-	# Remove punctuation
-	cleanedMessage = re.sub('([.,!?])','', cleanedMessage)
-	# Remove multiple spaces in message
-	cleanedMessage = re.sub(' +',' ', cleanedMessage)
-	return cleanedMessage
+    # Remove new lines within message
```
Comment (adeshpande3) [inline createDataset.py]: Thanks for fixing the tabs/spaces mixup!! So frustrating sometimes.

File: createDataset.py [hunk 6]
```diff
@@ -25,162 +151,74 @@ def getWhatsAppData():
...
-	combinedDictionary.update(getLinkedInData())
+    print 'Getting LinkedIn Data'
+    combinedDictionary.update(getLinkedInData())
 if (whatsAppData == 'y'):
-        print 'Getting whatsApp Data'
-        combinedDictionary.update(getWhatsAppData())
+    print 'Getting whatsApp Data'
+    combinedDictionary.update(getWhatsAppData())
+if (discordData == 'y'):
+    print 'Getting Discord Data'
```
Comment (adeshpande3) [inline createDataset.py]: Probably need parentheses around these prints? Or if you're saving the Python3 conversion changes for another PR, that's fine too.

File: createDataset.py [hunk 7]
```diff
@@ -25,162 +151,74 @@ def getWhatsAppData():
...
+                                # responses
+    for currentFile in allFiles:
+        with open(currentFile, 'r') as openedFile:
+            allLines = openedFile.readlines()
+        data = ''.join(allLines)
+        otherPersonsMessage, myMessage = "",""
+        response_sets = re.findall(r'\[.+\] (?!' + re.escape(personName) + r').+\n(.+)\n{2}(?:\[.+\] ' + re.escape(personName) + r'\n(.+)\n{2})', data)
+        for response_set in response_sets:
+            otherPersonsMessage = response_set[0]
+            myMessage = response_set[1]
```
Comment (adeshpande3) [inline createDataset.py]: I don't have as much experience with `re` but just wanted to make sure that `response_set` will always have at least 2 elements in it? If not, probably want some sort of check or graceful error handling.

File: createDataset.py [hunk 8]
```diff
@@ -4,18 +4,144 @@
...
 from datetime import datetime
 
-personName = raw_input('Enter your full name: ')
 fbData = raw_input('Do you have Facebook data to parse through (y/n)?')
 googleData = raw_input('Do you have Google Hangouts data to parse through (y/n)?')
 linkedInData = raw_input('Do you have LinkedIn data to parse through (y/n)?')
 whatsAppData = raw_input('Do you have whatsAppData to parse through (y/n)?')
+discordData = raw_input('Do you have discordData to parse through (y/n)?')
 
 def getWhatsAppData():
```
Comment (adeshpande3) [inline createDataset.py]: Just wanted to also confirm since the diff is a bit hard to parse. What were the changes to the existing functions? I see the `personName` change is one of them, but do you have a tl;dr of the other ones, if any?

File: createDataset.py (hunk 1 above)
Comment (TotallyNotChase) [reply createDataset.py]: precisely!

File: createDataset.py (hunk 6 above)
Comment (TotallyNotChase) [reply createDataset.py]: won't need parentheses in python 2 actually. This whole PR is still in python2, I'm working on porting to python 3 though.

File: createDataset.py (hunk 7 above)
Comment (TotallyNotChase) [reply createDataset.py]: yes if the match is *found*, it must have 2 elements. I've tested this one before, should be fine

File: createDataset.py (hunk 8 above)
Comment (TotallyNotChase) [reply createDataset.py]: All the changes I made, even the smallest ones, should be documented on the `Changelog` above. But yes the only diff to the other functions are, as you noticed, `personName` and also this
> Changed all open(...) commands in createDataset.py to with open(....)
Basically changed all the `openedFile = open(...)` to `with open(...) as openFile:`
It's for good practice, that's all.

**Comment by TotallyNotChase** [comment]
@adeshpande3 I replied to a bunch of the comments, always good to clear things up. I'll also fix the formatting on the comments real quick :)

File: createDataset.py (hunk 3 above)
Comment (TotallyNotChase) [reply createDataset.py]: The diff is being rendered wrong, that `str` cast is in line 157, `getLinkedinData`. It didn't have an `str` cast originally but that might be an oversight. I'll add it real quick

**Comment by TotallyNotChase** [comment]
Awesome! Ready for merge?
 If so let me know, I'll start an issue about the python2-python3 upgrade detailing my approach, it's almost done.

**Comment by adeshpande3** [comment]
Yeah go ahead and merge when you're ready
